packages/yoroi-extension/app/ergo-connector/stores/ConnectorStore.js

```diff
@@ -79,60 +80,45 @@ import type { IGetAllUtxosResponse } from '../../api/ada/lib/storage/models/Publ
 import type { IFetcher } from '../../api/ada/lib/state-fetch/IFetcher';
 import type { NetworkRow } from '../../api/ada/lib/storage/database/primitives/tables';
 
-// Need to run only once - Connecting wallets
-let initedConnecting = false;
-function sendMsgConnect(): Promise<ConnectingMessage> {
+export function connectorCall<T, R>(message: T): Promise<R> {
   return new Promise((resolve, reject) => {
-    if (!initedConnecting)
-      window.chrome.runtime.sendMessage((
-        { type: 'connect_retrieve_data' }: ConnectRetrieveData),
-        response => {
-          if (window.chrome.runtime.lastError) {
-            // eslint-disable-next-line prefer-promise-reject-errors
-            reject('Could not establish connection: connect_retrieve_data ');
-          }
-
-          resolve(response);
-          initedConnecting = true;
+    window.chrome.runtime.sendMessage(
+      message,
+      response => {
+        if (window.chrome.runtime.lastError) {
+          // eslint-disable-next-line prefer-promise-reject-errors
+          reject(`Could not establish connection: ${JSON.stringify(typeof message === 'object' ? message : '')}`);
```

Comment:
should `return` after `reject` because only either one of `reject` or `resolve` should be called.